taichi/transforms/make_adjoint.cpp

```diff
@@ -241,9 +241,8 @@ class MakeAdjoint : public IRVisitor {
     } else if (stmt->op_type == UnaryOpType::sqrt) {
       accumulate(stmt->operand,
                  mul(adjoint(stmt), div(constant(0.5f), sqrt(stmt->operand))));
-    } else if (stmt->op_type == UnaryOpType::cast) {
-      if (stmt->cast_by_value && is_real(stmt->cast_type) &&
-          is_real(stmt->operand->ret_type.data_type)) {
+    } else if (stmt->op_type == UnaryOpType::cast_value) {
+      if (is_real(stmt->cast_type) && is_real(stmt->operand->ret_type.data_type)) {
         accumulate(stmt->operand, adjoint(stmt));
       }
```

Comment:
I thought I found another potential BUG?
If `stmt->op_type = cast; stmt->cast_by_value = false`, then this `bit_cast` is simply ignored instead of throwing `TI_NOT_IMPLEMENTED`.

Reply:
That's actually fine. Bit cast generates 0 gradients. Feel free to leave a comment there though. Thanks!